### PR TITLE
Add option to pass kinds to AnnotateScmSlugEntityProcessor, eg ['Component', 'Template'] (#17351)

### PR DESCRIPTION
**Pull request by Pike**
## Hey, I just made a Pull Request!

Add option to pass kinds to AnnotateScmSlugEntityProcessor, like discussed in #17351 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Pike** [comment]
Ping.

**Comment by Pike** [comment]
Pong.

**Comment by freben** [comment]
Reopening!

**Comment by Pike** [comment]
Pung.

File: plugins/catalog-backend/src/modules/core/AnnotateScmSlugEntityProcessor.ts [hunk 1]
```diff
@@ -31,24 +31,31 @@ const GITLAB_ACTIONS_ANNOTATION = 'gitlab.com/project-slug';
...
+      kinds: kinds || ['Component'],
     });
   }
 
   async preProcessEntity(
     entity: Entity,
     location: LocationSpec,
   ): Promise<Entity> {
-    if (entity.kind !== 'Component' || location.type !== 'url') {
+    if (!this.opts.kinds.includes(entity.kind) || location.type !== 'url') {
```
Comment (Rugvip) [inline plugins/catalog-backend/src/modules/core/AnnotateScmSlugEntityProcessor.ts:58]: While we're at it I think it'd be best to make this comparison case insensitive

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice, thank you! 👍 

Just a minor nit

**Comment by kuangp** [comment]
I actually just implemented this in https://github.com/backstage/backstage/pull/18918 😄

**Pull request closed without merging**